lib/Model.php

```diff
@@ -1622,17 +1622,96 @@ public function __call($method, $args)
+
+        return $relation;
+    }
+
+    public static function join(string $joinStatement): Relation
+    {
+        $relation = new Relation(get_called_class(), static::$alias_attribute);
+        $relation->join($joinStatement);
+
+        return $relation;
+    }
+
+    public static function orderBy(string $order): Relation
+    {
+        $relation = new Relation(get_called_class(), static::$alias_attribute);
+        $relation->orderBy($order);
+
+        return $relation;
+    }
+
+    public static function groupBy(string $columns): Relation
+    {
+        $relation = new Relation(get_called_class(), static::$alias_attribute);
+        $relation->groupBy($columns);
+
+        return $relation;
+    }
+
+    public static function limit(int $limit): Relation
+    {
+        $relation = new Relation(get_called_class(), static::$alias_attribute);
+        $relation->limit($limit);
+
+        return $relation;
+    }
+
+    public static function offset(int $offset): Relation
+    {
+        $relation = new Relation(get_called_class(), static::$alias_attribute);
+        $relation->offset($offset);
+
+        return $relation;
+    }
+
+    public static function having(string $having): Relation
+    {
+        $relation = new Relation(get_called_class(), static::$alias_attribute);
+        $relation->having($having);
+
+        return $relation;
+    }
+
     /**
-     * Alias for self::find('all').
+     * needle is one of:
      *
-     * @see find
+     * primary key value        where(3)     WHERE author_id=3
+     * mapping of column names  where(["name"=>"Philip", "publisher"=>"Random House"]) finds the first row of WHERE name=Philip AND publisher=Random House
+     * raw WHERE statement      where(['name = (?) and publisher <> (?)', 'Bill Clinton', 'Random House'])
+     *
+     * @param int|string|array<string> $needle
      *
-     * @return array<int,static>
+     * @return Model|null The single row that matches query. If no rows match, returns null
      */
-    public static function all(/* ... */): array
+    public static function where(int|string|array $needle): Model|null
     {
-        /* @phpstan-ignore-next-line */
-        return static::find('all', ...func_get_args());
+        $relation = new Relation(get_called_class(), static::$alias_attribute);
+
+        return $relation->where($needle);
+    }
+
+    /**
+     * needle is one of:
+     *
+     * array of primary keys    all([1, 3, 5, 8]) WHERE author_id in (1, 3, 5, 8)
+     * mapping of column names  all(["name"=>"Philip", "publisher"=>"Random House"]) WHERE name=Philip AND publisher=Random House
+     * raw WHERE statement      all(['name = (?) and publisher <> (?)', 'Bill Clinton', 'Random House'])
+     *
+     * @param array<number|mixed|string> $needle An array containing values for the pk
+     *
+     * @return array<Model> All the rows that matches query. If no rows match, returns []
+     */
+    public static function all(array $needle = []): array
```

Comment:
`all` doesn't take any arguments; I think it's only meant as a means to get a handle to the `Relation` without executing anything:

```php
$allBooks = Books::all();

...
$books = $allBooks.where(['title'=>'foo'].to_a()'
```